### ISSUE TITLE
Не собирается библиотека под UE4

### ISSUE THREAD

**Issue by Mukudori** [gosha20777/keras2cpp#16]
Здравствуйте. Мне раньше не приходилось юзать СMake, поэтому существует вероятность, что я что-то делаю не так.

На хабре вы писали, что эта библиотека создавалась с целью быть использованной в движке UE4. У меня такая же цель. 

Во время сборки вот, что происходит в консоли:

> `D:\downloads\keras2cpp\build> cmake ..
> -- Configuring done
> -- Generating done
> -- Build files have been written to: D:/downloads/keras2cpp/build
> PS D:\downloads\keras2cpp\build> cmake --build .
> Microsoft (R) Build Engine версии 15.9.21+g9802d43bc3 для .NET Framework
> (C) Корпорация Майкрософт (Microsoft Corporation). Все права защищены.
> 
> cl : командная строка warning D9002: пропуск неизвестного параметра "-g" [D:\downloads\keras2cpp\build\keras2cpp.vcxpro
> j]
>   utils.cc
> cl : командная строка warning D9002: пропуск неизвестного параметра "-g" [D:\downloads\keras2cpp\build\keras2cpp.vcxpro
> j]
> cl : командная строка warning D9002: пропуск неизвестного параметра "-g" [D:\downloads\keras2cpp\build\keras2cpp.vcxpro
> j]
> D:\downloads\keras2cpp\src\utils.cc(6): warning C4834: отмена возвращаемого значения функции с атрибутом "nodiscard" [D
> :\downloads\keras2cpp\build\keras2cpp.vcxproj]
>   baseLayer.cc
>   activation.cc
>   conv1d.cc
>   conv2d.cc
>   dense.cc
>   elu.cc
>   embedding.cc
>   flatten.cc
>   lstm.cc
>   locally1d.cc
>   locally2d.cc
>   maxPooling2d.cc
>   batchNormalization.cc
>   model.cc
>   tensor.cc
>   cpp_model.cc
>   Создание кода...
>   keras2cpp.vcxproj -> D:\downloads\keras2cpp\build\Debug\keras2cpp.exe
> PS D:\downloads\keras2cpp\build> ./keras2cpp
> ./keras2cpp : Имя "./keras2cpp" не распознано как имя командлета, функции, файла сценария или выполняемой программы. Пр
> оверьте правильность написания имени, а также наличие и правильность пути, после чего повторите попытку.
> строка:1 знак:1
> + ./keras2cpp
> + ~~~~~~~~~~~
>     + CategoryInfo          : ObjectNotFound: (./keras2cpp:String) [], CommandNotFoundException
>     + FullyQualifiedErrorId : CommandNotFoundException`

Версия python = 3.7 (x64)
CMake 3.14

Что я делаю не так?

**Comment by Mukudori** [comment]
Попробовал просто скопировать содержимое папки scr в папку source проекта ue4, получил такой список ошибок при сборке 
![изображение](https://user-images.githubusercontent.com/23213509/63098098-4a11e180-bf9c-11e9-89ba-6a602666c90b.png)

![изображение](https://user-images.githubusercontent.com/23213509/63183201-a2b9ab00-c07e-11e9-9055-1f8df041b411.png)

**Comment by Mukudori** [comment]
Проблему частично решил, избавившись от темплейтов в классе Tensor и сделав его инициализацию строго через std::vector<size_t>. Пока с вашей моделью в примере работает под UE4 и отлично выводится результат в виджет. Но я планирую работать с LSTM сетями, поэтому подводные камни еще могут проявиться.

**Comment by gosha20777** [comment]
Вам надо было установить clang компилятор. g++ не работает)
